spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -587,18 +535,22 @@ def setup_menu(self):
         for name, func in functions:
             # QAction.triggered works differently for PySide and PyQt
             if not API == 'pyside':
-                slot = lambda _checked, func=func: self.change_type(func)
+                types_in_menu += [create_action(self, name,
+                                                triggered=lambda _checked,
+                                                func=func:
+                                                    self.change_type(func),
+                                                    context=Qt.WidgetShortcut)]
             else:
-                slot = lambda func=func: self.change_type(func)
-            types_in_menu += [create_action(self, name,
-                                            triggered=slot,
-                                            context=Qt.WidgetShortcut)]
+                types_in_menu += [create_action(self, name,
+                                                triggered=lambda func=func:
+                                                    self.change_type(func),
+                                                    context=Qt.WidgetShortcut)]

```

Comment:
Please simplify this if/else block by dropping the `pyside` part since we don't support Pyside anymore. Also, please leave things as they were, i.e.

```python
slot = lambda func=func: self.change_type(func)
types_in_menu += [create_action(self, name,
                                triggered=slot,
                                context=Qt.WidgetShortcut)
```

even if Pep8speaks disagrees with it.